packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -436,33 +443,52 @@ export async function createComponentTree({
     })(),
   }
 
+  // Create the node to render in this tree.
+  let node = (
+    <>
+      {isPage ? metadataOutlet : null}
+      {/* <Component /> needs to be the first element because we use `findDOMNode` in layout router to locate it. */}
+      {isPage && isClientComponent ? (
+        <StaticGenerationSearchParamsBailoutProvider
+          propsForComponent={props}
+          Component={Component}
+          isStaticGeneration={staticGenerationStore.isStaticGeneration}
+        />
+      ) : (
+        <Component {...props} />
+      )}
+      {/* This null is currently critical. The wrapped Component can render null and if there was not fragment
+          surrounding it this would look like a pending tree data state on the client which will cause an error
+          and break the app. Long-term we need to move away from using null as a partial tree identifier since it
+          is a valid return type for the components we wrap. Once we make this change we can safely remove the
+          fragment. The reason the extra null here is required is that fragments which only have 1 child are elided.
+          If the Component above renders null the actual tree data will look like `[null, null]`. If we remove the extra
+          null it will look like `null` (the array is elided) and this is what confuses the client router.
+          TODO-APP update router to use a Symbol for partial tree detection */}
+      {null}
+    </>
+  )
+
+  // If we're in static generation and we're forcing dynamic, we need to wrap
+  // the node in a component that will postpone rendering until after the
+  // initial render.
+  if (
+    staticGenerationStore.forceDynamic &&
+    staticGenerationStore.isStaticGeneration &&
+    staticGenerationStore.experimental.ppr
```

Comment:
Why is this a <MaybePostpone> if you already checked the conditions? `maybePostpone` checks the same things again.

It looks to me like the only special thing `maybePostpone` does is set a special field on the static generation store. But this is fishy to me because don’t we need to bail out if React.postpone is called in userspace, too? So this suggests to me we’re either doing something redundant here or we’re missing logic elsewhere to handle the userspace postpone case.